src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/TypeCast.cs

```diff
@@ -475,6 +475,10 @@ private static unsafe object CheckCastClassSpecial(MethodTable* pTargetType, obj
             return ThrowInvalidCastException(pTargetType);
         }
 
+        [RuntimeExport("RhTypeCast_IsAssignable")]
+        public static unsafe bool IsAssignable(MethodTable* fromTypeMT, MethodTable* toTypeMT)
+            => AreTypesAssignable(fromTypeMT, toTypeMT);
+
```

Comment:
The method this is calling into is already exposed as the `RhTypeCast_AreTypesAssignable` RuntimeExport. Do we need a new RuntimeExport? Does this API have the expected semantics? (`AreTypesAssignable` assumes a boxed source so if `fromTypeMT` is a struct and `toTypeMT` is System.Object or System.ValueType, it will return true.)

Reply:
The semantics here appear correct. I've added tests for the boxing source semantics.